internal/logic/logic_test.go

```diff
@@ -319,10 +321,282 @@ func TestTransactionManagementServiceLogic_NewTransaction(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			rec := NewTransactionManagementServiceLogic(tt.setup())
-			time.Sleep(2 * time.Second)
+
 			got := rec.NewTransaction(tt.credentials)
 
 			tt.want(got)
 		})
 	}
 }
+
+func TestTransactionManagementServiceLogic_DownloadTransactions(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tests := []struct {
+		name        string
+		credentials string
+		setup       func() (datasource.DataSourceI, config.ExternalSvc)
+		want        func(*respModel.Response)
+	}{
+		{
+			name:        "Success :: DownloadPdf",
+			credentials: "123",
+			setup: func() (datasource.DataSourceI, config.ExternalSvc) {
+				mockDs := mock.NewMockDataSourceI(mockCtrl)
+				var trans []model.Transaction
+				trans = append(trans, model.Transaction{UserId: "123", AccountNumber: 1})
+				mockDs.EXPECT().Get(map[string]interface{}{"transaction_id": "123"}, 0, 0).Times(1).Return(trans, 1, nil)
+				router := mux.NewRouter()
+				router.HandleFunc("/microbank/v1/user", func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "Success", map[string]interface{}{"Name": "abc"})
+				})
+				mockPdf := pdfMock.NewMockHtmlToPdfSvcI(mockCtrl)
+				mockPdf.EXPECT().GeneratePdf(gomock.Any(), gomock.Any()).Return([]byte("PDF"), nil)
+				srv := httptest.NewServer(router)
+				return mockDs, config.ExternalSvc{UserSvc: srv.URL, PdfSvc: config.PdfSvc{UuId: "11-22-33-44", PdfService: mockPdf}}
+			},
+			want: func(resp *respModel.Response) {
+				temp := respModel.Response{
+					Status:  http.StatusOK,
+					Message: "SUCCESS",
+					Data:    []byte("PDF"),
+				}
+				if !reflect.DeepEqual(resp.Status, temp.Status) {
```

Comment:
What is the issue on comparing the entire struct instead of comparing every field separately? It looks like this is done for multiple test cases.

Reply:
done